### PR TITLE
Fix tests on circle

### PR DESCRIPTION
**Pull request by algrs**
But does it? Will we ever know?

### CODE REVIEW

File: baiji/test_s3.py [hunk 1]
```diff
@@ -26,9 +26,8 @@ def setUp(self):
         self.s3_path = loc.path
 
         b = s3.S3Connection()._bucket(self.bucket) # pylint: disable=protected-access
-        for key in b.list(self.s3_path[1:]):
-            b.delete_key(key)
-        self.assertEqual(len([x for x in b.list(self.s3_path[1:])]), 0) # just make sure we're starting with a clean slate
+        b.delete_key(self.s3_path[1:] + '.tempdir')
```
Comment (lillekemiker) [inline baiji/test_s3.py]: What's the logic behind this? Do we know that this is the only file that might be there?

Comment (algrs) [inline baiji/test_s3.py]: Yeah, that file is written by `s3.path.gettmpdir()` and is guaranteed to be the only file in the directory. The old "iterate and delete all" code dated to a version of this that was sharing the temp dir amongst all the tests.

**Comment by lillekemiker** [comment]
Okay, I am satisfied. I rolled the die thrice without failures, so clearly the transient error has been fixed. Merge away 👍

**Comment by chenliu0831** [comment]
Merge this?

**Pull request merged**